examples/nidaqmx_analog_input/measurement.py

```diff
@@ -30,31 +44,93 @@ def measure(physical_channel, sample_rate, number_of_samples):
+                context=measurement_service.context.pin_map_context,
+                pin_or_relay_names=[physical_channel],
+                instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DAQMX,
+                # If another measurement is using the session, wait for it to complete.
+                # Specify a timeout to aid in debugging missed unreserve calls.
+                # Long measurements may require a longer timeout.
+                timeout=60,
+            )
+        )
+        if len(reservation.session_info) != 1:
+            measurement_service.context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"Unsupported number of sessions: {len(reservation.session_info)}",
+            )
+        session_info = reservation.session_info[0]
+        task = stack.enter_context(_create_nidaqmx_task(session_info))
+        channel_mappings = session_info.channel_mappings
+        pending_cancellation = False
+
+        def cancel_callback():
+            logging.info("Canceling measurement")
+            session_to_abort = task
+            task_to_abort = task
+            if task_to_abort is not None:
+                nonlocal pending_cancellation
+                pending_cancellation = True
+                session_to_abort.abort()
+
+        measurement_service.context.add_cancel_callback(cancel_callback)
+        time_remaining = measurement_service.context.time_remaining
+
+        timeout = min(time_remaining, 10.0)
+        channel = next(
+            channel_mapping.channel
+            for channel_mapping in channel_mappings
+            if channel_mapping.pin_or_relay_name == physical_channel
+        )
+        task.ai_channels.add_ai_voltage_chan(channel, min_val=0, max_val=5)
         task.timing.cfg_samp_clk_timing(
             rate=sample_rate,
             samps_per_chan=number_of_samples,
         )
-        voltage_values = task.read(number_of_samples_per_channel=number_of_samples, timeout=timeout)
+        voltage_values = task.read(
+            number_of_samples_per_channel=number_of_samples, timeout=timeout
+        )
         task = None  # Don't abort after this point
 
     _log_measured_values(voltage_values)
     logging.info("Completed measurement")
     return (voltage_values,)
 
 
+def _create_nidaqmx_task(
+    session_info: nims.session_management.SessionInformation,
+) -> nidaqmx.Task:
+    if service_options.use_grpc_device:
+        session_grpc_address = service_options.grpc_device_address
+
+        if not session_grpc_address:
+            session_grpc_channel = measurement_service.get_channel(
+                provided_interface=GRPC_SERVICE_INTERFACE_NAME,
+                service_class="ni.measurementlink.v1.grpcdeviceserver",
+            )
+        else:
+            session_grpc_channel = measurement_service.channel_pool.get_channel(
+                target=session_grpc_address
+            )
+        grpc_session_options = GrpcSessionOptions(
```

Comment:
Ok. This change is already present in nidaqmx example's **pyproject.toml**.
And also, I have uncommented it as suggested.
https://github.com/ni/measurementlink-python/blob/cea81e5d29c216ad12a2ee839288b84632a094c5/examples/nidaqmx_analog_input/pyproject.toml#LL13C1-L13C1

Reply:
On further thought, a version check would also make sense, once nidaqmx-python's version has been updated: https://github.com/ni/nidaqmx-python/pull/335

```
[tool.poetry.dependencies]
python = "^3.8"
nidaqmx = { version = ">=0.8.0-dev0", allow-prereleases = true }
ni-measurementlink-service = "^1.0.0"
click = ">=7.1.2"

[tool.poetry.group.dev.dependencies]
mypy = ">=1.0"
# Uncomment to use prerelease dependencies (requires poetry>=1.2).
# TODO: Comment out once a prerelease is built.
nidaqmx = { git = "https://github.com/ni/nidaqmx-python.git", branch = "users/bkeryan/update-version" }
# ni-measurementlink-service = {path = "../..", develop = true}
```

Reply:
Sure. Since the relevant PR in nidaqmx-python has already been merged to master and branch has already been deleted, I have referred to the branch `master` in **pyproject.toml**
https://github.com/ni/nidaqmx-python/pull/335